packages/manager/apps/web/client/app/hosting/hosting.routes.js

```diff
@@ -58,10 +58,45 @@ export default /* @ngInject */ ($stateProvider) => {
           ),
       pendingTasks: /* @ngInject */ (HostingTask, serviceName) =>
         HostingTask.getPending(serviceName).catch(() => []),
+      privateDatabasesIds: /* @ngInject */ (HostingDatabase, serviceName) =>
+        HostingDatabase.getPrivateDatabaseIds(serviceName),
+      privateDatabasesDetachable: /* @ngInject */ (
+        $q,
+        ovhManagerProductOffersDetachService,
+        PrivateDatabase,
+        privateDatabasesIds,
+      ) =>
+        $q
+          .all(
+            privateDatabasesIds.map((id) =>
+              PrivateDatabase.getServiceInfos(id).catch(() => null),
+            ),
+          )
+          .then((privateDatabasesInformation) =>
+            privateDatabasesInformation
+              .filter((information) => information !== null)
+              .flatten(),
+          )
+          .then((privateDatabasesInformation) =>
+            $q.all(
+              privateDatabasesInformation.map(({ domain, serviceId }) =>
+                ovhManagerProductOffersDetachService
+                  .getAvailableDetachPlancodes(serviceId)
+                  .catch(() => [])
+                  .then((plancodes) => ({
+                    optionId: domain,
+                    serviceId,
+                    detachPlancodes: plancodes,
+                  })),
+              ),
+            ),
+          ),
```

Comment:
Maybe we can add a `catch` to avoid crashing the whole hosting dashboard and adapt the display inside ? (can be done in another PR as this is already QC ok)

Reply:
If I'm not wrong, L85 does the job to secure the call, as it is the only API request done in this resolve :).
But you make me think of securing the call to retrieve private databases for a hosting.
And with all those issues we had with hosting, I'll do it in this PR